packages/worksheet/src/components/WorksheetBox/index.js

```diff
@@ -1,22 +1,128 @@
-import { IconByName, H2, Caption, Subtitle } from "@shiksha/common-lib";
 import {
-  Avatar,
-  Box,
-  HStack,
-  Pressable,
-  Stack,
-  Text,
-  VStack,
-} from "native-base";
+  capture,
+  IconByName,
+  likeRegistryService,
+  telemetryFactory,
+  H2,
+  Caption,
+  Subtitle,
+  commentRegistryService,
+} from "@shiksha/common-lib";
+import { Avatar, Box, HStack, Pressable, Stack, VStack } from "native-base";
 import React from "react";
 import { useTranslation } from "react-i18next";
 import { useNavigate } from "react-router-dom";
+import jwt_decode from "jwt-decode";
 
-export default function WorksheetBox({ item, url, canShare, _addIconButton }) {
+export default function WorksheetBox({
+  item,
+  url,
+  canShare,
+  canShowButtonArray,
+  _addIconButton,
+  appName,
+}) {
   const { t } = useTranslation();
   const navigate = useNavigate();
   const colors = ["lightBlue.800", "indigo.900", "fuchsia.700", "rose.600"];
-  const random = Math.floor(Math.random() * (4 - 1) + 1) - 1;
+  const [like, setLike] = React.useState({});
+  const [likes, setLikes] = React.useState([]);
+  const [comments, setComments] = React.useState([]);
+  const [random, setRandom] = React.useState();
+  const { sub } = jwt_decode(localStorage.getItem("token"));
+
+  React.useEffect(async (e) => {
+    setRandom(Math.floor(Math.random() * (4 - 1) + 1) - 1);
+    await getLikes();
+    await getComments();
+  }, []);
+
+  const getLikes = async () => {
+    const result = await likeRegistryService.getAll({
+      contextId: { eq: item?.id },
+      context: { eq: "Worksheet" },
+      type: { eq: "like" },
+    });
+    const newData = result.find((e, index) => e.userId === sub);
+    setLikes(result ? result : []);
+    setLike(newData ? newData : {});
+  };
+
+  const getComments = async () => {
```

Comment:
move this logic to worksheet service.

there can be a function like getWorksheetCmments(worksheetIId) which abstract context query